### DISCUSSION TITLE
useRoute() returning the first route I accessed instead of the current route

### DISCUSSION THREAD

**Discussion by leonardoMetaway** [nuxt/nuxt#23343] [Questions]
whenever I access useRoute() in a page, instead of getting the current page I instead get the first page I accessed, for example:
 I first access my project from the url `http://localhost:3000/analises` and then use `navigateTo(/analise/[codigo])` where I have useRoute():
 
![image](https://github.com/nuxt/nuxt/assets/143647557/d67b27a6-206a-4f7e-bb69-3eba4e1a0bfb)
If I first accessed it from `http://localhost:3000` it would console.log `'/'` instead:
![image](https://github.com/nuxt/nuxt/assets/143647557/a189aff0-198e-4d25-b17c-8b21c8da2d32)

Using useRouter() does return the correct currentRoute:
![image](https://github.com/nuxt/nuxt/assets/143647557/36b8ba25-70e3-49ad-896b-b4c44e1f7829)
Viewing the $route on vue devtools does return the correct route: 
![image](https://github.com/nuxt/nuxt/assets/143647557/75310663-6b5c-4cef-a780-873dfce3de0e)

I tried debugging a few things, such as removing the useRoute() everywhere else, deleting my routing middleware, updating nuxt to 3.7.3 but I still have no clue what is going on... Does anybody have a clue what I might have done wrong that resulted in this?

**Comment by xavier-ottolini** [comment]
Hi!

I have got the same issue. Did you fix it ? How ?

**Reply by Leo-Felde** [reply]
I think we are just supposed to use useRouter() instead. I couldn't find any solution

**Comment by xjccc** [comment]
https://github.com/nuxt/nuxt/issues/21340 something like this?

**Reply by Leo-Felde** [reply]
I see that it makes sense to happen if I use it inside the layout, however the issue seems to persist if you use it inside a page component also. It would sorta make sense if the page was pre-rendered I guess but that is not my case....
